packages/flutter/test/rendering/layers_test.dart

```diff
@@ -239,4 +290,169 @@ void main() {
+    //
+    // ────────────────────────────
+    test('Non-overlapping layers at wrong elevation', () {
+      final PhysicalModelLayer layerA = PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(0, 0, 20, 20)),
+        elevation: 3.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      );
+      final PhysicalModelLayer layerB =PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(20, 20, 20, 20)),
+        elevation: 2.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      );
+      _testConflicts(layerA, layerB, 0);
+    });
+
+    // Tests:
+    //
+    //     ───────                       (Child of A, paints second)
+    //   ───────────                     (LayerA, paints first)
+    //                ────────────       (LayerB, paints third)
+    //
+    // ────────────────────────────
+    test('Non-overlapping layers at wrong elevation, child at lower elevation', () {
+      final PhysicalModelLayer layerA = PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(0, 0, 20, 20)),
+        elevation: 3.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      );
+
+      layerA.append(PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(2, 2, 10, 10)),
+        elevation: 1.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      ));
+
+      final PhysicalModelLayer layerB =PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(20, 20, 20, 20)),
+        elevation: 2.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      );
+      _testConflicts(layerA, layerB, 0);
+    });
+
+    // Tests:
+    //
+    //          ─────────                (Child of A, paints second, overflows)
+    //                ────────────       (LayerB, paints third)
+    //   ───────────                     (LayerA, paints first)
+    //
+    //
+    // ────────────────────────────
+    //
+    // Which fails because the overflowing child overlaps something that paints
+    // after it at a lower elevation.
+    test('Child overflows parent and overlaps another physical layer', () {
+      final PhysicalModelLayer layerA = PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(0, 0, 20, 20)),
+        elevation: 3.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      );
+
+      layerA.append(PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(15, 15, 25, 25)),
+        elevation: 2.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      ));
+
+      final PhysicalModelLayer layerB =PhysicalModelLayer(
+        clipPath: Path()..addRect(Rect.fromLTWH(20, 20, 20, 20)),
+        elevation: 4.0,
+        color: const Color(0),
+        shadowColor: const Color(0),
+      );
+
+      _testConflicts(layerA, layerB, 1);
+    });
+  });
```

Comment:
What about:

````
       -----------------     paints third
----------             |          paints forth
    |                 ---------- paints second
    |                        |
    |                        |
----------------------------------- paints first
````

Reply:
Ignore this one. It's the same as above. Maybe add `|` child indicators.

Reply:
This is covered - making diagram clearer